apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -0,0 +1,374 @@
+                    </div>
+                    <div style={{fontFamily: '"Gotham 3r", sans-serif', fontSize: 12, paddingBottom: 20}}>
+                      {getTagString("grade", this.props.item.tags_grade)} | {getTagString("programming_language", this.props.item.tags_programming_language)}
+                    </div>
+                    <div style={{fontFamily: '"Gotham 3r", sans-serif', fontSize: 14}}>
+                      {this.props.item.longdescription}
+                    </div>
+                  </div>
+                  <div style={{clear: 'both'}}/>
+                  <table style={{marginTop: 20, width: '100%'}}>
+                    <tbody>
+                      {tableEntries.map(item =>
+                        <tr key={item.key}>
+                          <td style={{padding: 5, width: '40%', fontFamily: '"Gotham 5r", sans-serif', border: '1px solid lightgrey'}}>
+                            {item.title}
+                          </td>
+                          <td style={{padding: 5, border: '1px solid lightgrey'}}>
+                            {item.body}
+                          </td>
+                        </tr>
+                      )}
+                    </tbody>
+                  </table>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+  const Tutorial = React.createClass({
+    propTypes: {
+      item: React.PropTypes.object.isRequired
+    },
+
+    getInitialState: function () {
+      return {
+        showingDetail: false
+      };
+    },
+
+    tutorialClicked: function () {
+      this.setState({showingDetail: true});
+    },
+
+    tutorialDetailClosed: function () {
+      this.setState({showingDetail: false});
+    },
+
+    render() {
+      return (
+        <div>
+          <TutorialDetail showing={this.state.showingDetail} item={this.props.item} closeClicked={this.tutorialDetailClosed}/>
+          <div className="col-33" style={{float: 'left', padding: 2}}>
+            <div style={{padding: 5}}>
+              <div style={{cursor: 'pointer'}} onClick={this.tutorialClicked}>
+                <img src={this.props.item.image} style={{width: '100%', height: 180}}/>
+                <div style={{fontFamily: '"Gotham 5r", sans-serif', fontSize: 15, textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden'}}>
+                  {this.props.item.name}
+                </div>
+                <div style={{fontFamily: '"Gotham 3r", sans-serif', fontSize: 12, paddingBottom: 20}}>
+                  {getTagString("grade", this.props.item.tags_grade)} | {getTagString("programming_language", this.props.item.tags_programming_language)}
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+  const TutorialSet = React.createClass({
+    propTypes: {
+      tutorials: React.PropTypes.arrayOf(
+        React.PropTypes.object
```

Comment:
It looks like we depend on this object being of a particular shape. Can we describe this using `React.PropTypes.shape`?